src/solver/activ/fwd_1.cpp

```diff
@@ -0,0 +1,274 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/activ/solvers.hpp>
+
+#include <miopen/activ/invoke_params.hpp>
+#include <miopen/activ/problem_description.hpp>
+#include <miopen/visit_float.hpp>
+#include <miopen/kernel_build_params.hpp>
+
+namespace miopen {
+
+namespace solver {
+
+namespace activ {
+
+bool ActivFwdSolver1::IsApplicable(const ExecutionContext& context,
+                                   const miopen::activ::ProblemDescription& problem) const
+{
+    if(problem.GetDirection() != miopen::activ::Direction::Forward)
+        return false;
+
+    const auto x_elem_sz = problem.GetXDesc().GetElementSize();
+    const auto y_elem_sz = problem.GetYDesc().GetElementSize();
+
+    if(x_elem_sz != y_elem_sz)
+        return false;
+
+    // Todo: probably fix "the rest" logic here
+    return !ActivFwdSolver0{}.IsApplicable(context, problem);
+}
+
+ConvSolution ActivFwdSolver1::GetSolution(const ExecutionContext&,
+                                          const miopen::activ::ProblemDescription& problem) const
+{
+    const auto& xDesc = problem.GetXDesc();
+    const auto& yDesc = problem.GetYDesc();
+
+    int nOut       = 1;
+    int cOut       = 1;
+    int hOut       = 1;
+    int wOut       = 1;
+    int nOutStride = 0;
+    int cOutStride = 0;
+    int hOutStride = 0;
+    int wOutStride = 0;
+
+    if(yDesc.GetSize() == 4)
+    {
+        std::tie(nOut, cOut, hOut, wOut)                         = tien<4>(yDesc.GetLengths());
+        std::tie(nOutStride, cOutStride, hOutStride, wOutStride) = tien<4>(yDesc.GetStrides());
+    }
+    else if(yDesc.GetSize() < 4 && yDesc.GetSize() > 0)
+    {
+        auto tensor_size = yDesc.GetSize();
+        switch(tensor_size)
+        {
+        case 1:
+            std::tie(wOut)       = tien<1>(yDesc.GetLengths());
+            std::tie(wOutStride) = tien<1>(yDesc.GetStrides());
+            nOutStride           = wOut * wOutStride;
+            cOutStride           = wOut * wOutStride;
+            hOutStride           = wOut * wOutStride;
+            break;
+        case 2:
+            std::tie(hOut, wOut)             = tien<2>(yDesc.GetLengths());
+            std::tie(hOutStride, wOutStride) = tien<2>(yDesc.GetStrides());
+            nOutStride = hOut * hOutStride;
+            cOutStride = hOut * hOutStride;
+            break;
+        case 3:
+            std::tie(cOut, hOut, wOut)                   = tien<3>(yDesc.GetLengths());
+            std::tie(cOutStride, hOutStride, wOutStride) = tien<3>(yDesc.GetStrides());
+            nOutStride = cOut * cOutStride;
+            break;
+        default: assert(false);
+        }
+    }
+    else
+    {
+        MIOPEN_THROW("activation does not support tensor size larger than 4 or smaller than 1");
```

Comment:
`...miopenStatusNotImplemented,...`

Reply:
...Or this is just a move of some existing code?

Reply:
Existing, but I'll change anyway.

Reply:
Thanks!